File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/recursion.java
/**
 * Java Implementation: Recursion
 * Invariant: Call Stack O(Depth), Work O(Branch^Depth)
 */
package seed.dsa.core;

import java.util.*;

public class Recursion<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/javascript/recursion.js
/**
 * JavaScript Implementation: Recursion
 * Invariant: Call Stack O(Depth), Work O(Branch^Depth)
 */
export class Recursion {
  constructor() {
    this.items = [];
  }

  // TODO: Implement verified invariant
  insert(val) {
    this.items.push(val);
  }

  get size() {
    return this.items.length;
  }
}
